app/controllers/works_controller.rb

```diff
@@ -0,0 +1,95 @@
+class WorksController < ApplicationController
+
+    def index
+        @works = Work.all
+    end
+
+    def show
+        @work = Work.find_by(id: params[:id])
+
+        if @work.nil?
+            head :not_found
+            return
+        end 
```

Comment:
Note this functionality is used in multiple methods. This is a great place for a controller filter `find_work`